cartographer_ros/cartographer_ros/node.cc

```diff
@@ -118,7 +118,7 @@ Node::Node(const NodeOptions& node_options, tf2_ros::Buffer* const tf_buffer)
       &Node::PublishConstraintList, this));
 }
 
-Node::~Node() {}
+Node::~Node() { FinishAllTrajectories(); }
```

Comment:
This could be added before `return`ing in  `ff (!ros::ok())`, but I thought it is more appropriate to add it here.

Reply:
We [did](https://github.com/googlecartographer/cartographer_ros/blob/f61e513bd6479895041a9f2918505250b5bb1eb9/cartographer_ros/cartographer_ros/offline_node.cc#L229). It is and was mandatory, or you get a failed check at exiting. Previously, we would always execute that line, even if we did Ctrl-C. In this implementation, finalizing is tied to encountering the last message, which obviously will not happen when you do Ctrl-C.

It's just that I think node should handle it for us.